package/com.unity.formats.usd/Runtime/Scripts/IO/Materials/ShaderExporterBase.cs

```diff
@@ -33,33 +33,120 @@ public class ShaderExporterBase {
+              textureIsExported = true;
+          }
+        }
+      }
+  #endif
+      if (!textureIsExported)
+      {
+        fileName = texture.name + "_" + Random.Range(10000000, 99999999).ToString();
+  #if UNITY_EDITOR
+        if (UnityEditor.AssetDatabase.Contains(texture))
+          fileName = texture.name + "_png";
+  #endif
+        filePath = System.IO.Path.Combine(destTexturePath, fileName + ".png");
+        
+        // TODO extra care has to be taken of Normal Maps etc., since these are in a converted format in memory (namely, 16 bit G and A instead of 8 bit RGBA)
+        // An example of this conversion in a shader is in Khronos' UnityGLTF implementation.
+        // basically, the blit has do be done with the right unlit conversion shader to get a proper "file-based" tangent space normal map back
+
+        // Blit the texture and get it back to CPU
+        // Note: Can't use RenderTexture.GetTemporary because that doesn't properly clear alpha channel
+        var rt = new RenderTexture(texture.width, texture.height, 0, RenderTextureFormat.ARGB32);
+        var resultTex = new Texture2D(texture.width, texture.height, TextureFormat.ARGB32, true);
+        try { 
+          Graphics.Blit(texture, rt);
+          RenderTexture.active = rt;
+          resultTex.ReadPixels(new Rect(0, 0, texture.width, texture.height), 0, 0);
+          resultTex.Apply();
+
+          System.IO.File.WriteAllBytes(filePath, resultTex.EncodeToPNG());
+          if (System.IO.File.Exists(filePath))
+            textureIsExported = true;
+        }
+        finally {
+          RenderTexture.active = null;
+          rt.Release();
+          GameObject.DestroyImmediate(rt);
+          GameObject.DestroyImmediate(resultTex);
+        }
+      }
+
+      if(!textureIsExported)
+      {
+        var tex = new Texture2D(1, 1, TextureFormat.ARGB32, true);
+        try { 
+          tex.SetPixel(0, 0, Color.white);
+          tex.Apply();
+          System.IO.File.WriteAllBytes(filePath, tex.EncodeToPNG());
+          if (System.IO.File.Exists(filePath))
+              textureIsExported = true;
+        }
+        finally
+        {
+          GameObject.DestroyImmediate(tex);
+        }
+      }
+
+      if(textureIsExported)
+      { 
+        // Make file path baked into USD relative to scene file and use forward slashes.
+        filePath = ImporterBase.MakeRelativePath(scene.FilePath, filePath);
+        filePath = filePath.Replace("\\", "/");
+
+        var uvReader = new PrimvarReaderSample<Vector2>();
+        uvReader.varname.defaultValue = new TfToken("st");
+        scene.Write(usdShaderPath + "/uvReader", uvReader);
+        var tex = new TextureReaderSample(filePath, usdShaderPath + "/uvReader.outputs:result");
+        scene.Write(usdShaderPath + "/" + textureName, tex);
+        return usdShaderPath + "/" + textureName + ".outputs:" + textureOutput;
+      }
+      else
+        throw new System.Exception("Texture wasn't exported.");
```

Comment:
Nit: add braces.

Question: Is this truly an exception or do we expect it to happen sometimes?

Reply:
This was just copied over from the previous implementation - I changed it to an error log and added some more sane help to the user what actually went wrong.

Reply:
It was happening all the time before my changes, now it should happen much less often:
- if the texture is null
- if it's a 3D texture that can't be blitted
- something's wrong on disk write (e.g. a known issue of mine is that if the export path is exactly where the src path is this will obviously fail but could be handled better - not covered by this PR).